proxy/ProxySession.h

```diff
@@ -143,6 +145,14 @@ class ProxySession : public VConnection
 
   APIHook *hook_get(TSHttpHookID id) const;
   HttpAPIHooks const *feature_hooks() const;
+
+  // Returns null pointer if session does not use a TLS connection.
+  SSLProxySession const *
+  ssl() const
+  {
+    return _ssl.get();
+  }
```

Comment:
could you move the function definition to the bottom of this .h file.  Taming this file is difficult.